src/flask/app.py

```diff
@@ -1256,11 +1260,11 @@ def index():
             )
 
         if provide_automatic_options is None:
-            if "OPTIONS" not in methods:
+            if "OPTIONS" in methods:
+                provide_automatic_options = False
+            else:
```

Comment:
why not 
`provide_automatic_options = False if "OPTIONS" in methods else True`

Reply:
Because there's other stuff going on in the `else` block.

Reply:
Absolutely not. First of all, this is just adding more complexity than needed as your code is just a more convoluted way of this:

    provide_automatic_options = "OPTIONS" not in methods

But then, if you look a the whole code, you'll notice that the code in the else branch does more than just setting the variable to True...